dask/tests/test_delayed.py

```diff
@@ -712,3 +713,26 @@ def test_dask_layers_to_delayed():
     assert d.dask.layers.keys() == {"delayed-" + name}
     assert d.dask.dependencies == {"delayed-" + name: set()}
     assert d.__dask_layers__() == ("delayed-" + name,)
+
+
+def test_annotations_survive_optimization():
+    with dask.annotate(foo="bar"):
+        graph = HighLevelGraph.from_collections(
+            "b",
+            {"a": 1, "b": (inc, "a"), "c": (inc, "b")},
+            [],
+        )
+        d = Delayed("b", graph)
+
+    assert type(d.dask) is HighLevelGraph
+    assert len(d.dask.layers) == 1
+    assert len(d.dask.layers["b"]) == 3
+    assert next(iter(d.dask.layers.values())).annotations == {"foo": "bar"}
```

Comment:
Nitpick: I could be missing something, but isn't this equivalent? (If so, I'd claim it's easier for readers to reason about)

```suggestion
    assert d.dask.layers["b"].annotations == {"foo": "bar"}
```

Reply:
Yes, that is easier, I think the prior implementation came about when this test had a less predictable layer name.

Reply:
Fixed